ComputerInterface/BaseGameInterface.cs

```diff
@@ -44,51 +42,113 @@ public static Color GetColor()
+            if (CheckForComputer(out var computer) && PhotonNetwork.InRoom && GorillaTagger.Instance.myVRRig != null)
+                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, r, g, b, computer.leftHanded);
+        }
 
-            if (!GorillaComputer.instance.CheckAutoBanListForName(name)) return;
+        public static void InitializeNoobMaterial(Color color)
+        {
+            if (CheckForComputer(out var computer) && PhotonNetwork.InRoom && GorillaTagger.Instance.myVRRig != null)
+                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, color.r, color.g, color.b, computer.leftHanded);
+        }
 
-            if (name.Length > MAX_NAME_LENGTH)
-            {
-                name = name.Substring(0, MAX_NAME_LENGTH);
-            }
+        #endregion
 
-            PhotonNetwork.LocalPlayer.NickName = name;
-            GorillaComputer.instance.offlineVRRigNametagText.text = name;
-            GorillaComputer.instance.savedName = name;
-            PlayerPrefs.SetString("playerName", name);
-            PlayerPrefs.Save();
+        #region Name settings
 
-            GetColor(out var r, out var g, out var b);
-            InitializeNoobMaterial(r, g, b);
+        public static string GetName()
+        {
+            if (CheckForComputer(out var computer)) return computer.savedName;
+            return null;
         }
 
-        public static void InitializeNoobMaterial(float r, float g, float b) => InitializeNoobMaterial(new Color(r, g, b));
-
-        public static void InitializeNoobMaterial(Color color)
+        public static void SetName(string name, out bool isError, out string errorReason)
         {
-            if (PhotonNetwork.InRoom)
-                GorillaTagger.Instance.myVRRig.photonView.RPC("InitializeNoobMaterial", RpcTarget.All, color.r, color.g, color.b, GorillaComputer.instance?.leftHanded ?? true);
+            if (CheckForComputer(out var computer))
+            {
+                if (!NameAllowed(name, out string errorReasonInit))
+                {
+                    isError = true;
+                    errorReason = errorReasonInit;
+                    return;
+                }
+
+                try
+                {
+                    name = name.Replace(" ", "");
+                    computer.currentName = name;
+                    PhotonNetwork.LocalPlayer.NickName = name;
+                    computer.offlineVRRigNametagText.text = name;
+                    computer.savedName = name;
+                    PlayerPrefs.SetString("playerName", name);
+                    PlayerPrefs.Save();
+
+                    GetColor(out var r, out var g, out var b);
+                    InitializeNoobMaterial(r, g, b);
+                }
+                catch
+                {
+                    throw new NullReferenceException(); // usually occurs due to the InitializeNoobMaterial method
+                }
```

Comment:
Since another exception is immediately thrown, does this even do anything?

Reply:
Ah that, I forgot to remove it when I did fix the method mentioned in the comment

Reply:
Usually when it would falsely check to see if it was or wasn't null I don't remember